### ISSUE TITLE
Bug: `TSImportType`'s `option` object always has key `with`

### ISSUE THREAD

**Issue by overlookmotel** [typescript-eslint/typescript-eslint#11114]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&showAST=es&fileType=.tsx&code=C4TwDgpgBAKlC8UCWBbMB7ATsAFAIhTwBooBvKAdyWAAsAuMqPTCAZ3QBsBXYJdAOwC0KdABMIeBnlQZseKAF9FASgDcAKFCQoAVQTI0WXAWKMAhq1YRsDcszacefISPGSmMo-KUK16oA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type T = import("m", { with: { "resolution-mode": "import" } });
type U = import("m", { assert: { "resolution-mode": "import" } });
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

Both statements include a `TSImportType` with an `options` property containing an `ObjectExpression`. Both `ObjectExpression`s have a single `Property`.

The key of `Property` in 1st statement is `with`.

The key of `Property` in 2nd statement is also `with`! I'd have expected it to be `assert`.

### Actual Result

(see above)

### Additional Info

I *think* this is a bug. But apologies if this is a stupid query. I don't know TypeScript at all well.

I notice that TypeScript's AST doesn't seem to differentiate between the two either, so maybe it's not possible for TS-ESLint to?

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/parser`            | `8.31.1` |

**Comment by overlookmotel** [comment]
Actually, there is a way to differentiate between the two - the `range` on the `Identifier` for the property key has length 6 when it's `assert`, vs 4 when it's `with`.

That's a pretty odd workaround, but as far as I'm aware, `with` and `assert` are the only legal values in this position, so it should be accurate.

**Comment by bradzacher** [comment]
the reason this happens is because TS doesn't differentiate between the two and our code just blindly grabs the token without checking its text -- which is why the range is correct but the name is not.

https://github.com/typescript-eslint/typescript-eslint/blob/50ef8d9b50299d0367e707a7c8e466053af512f6/packages/typescript-estree/src/convert.ts#L3121-L3139

**Comment by overlookmotel** [comment]
Thanks very much for coming back so swiftly. PR to fix: #11115.
